### PR TITLE
Options menu and gameplay tweaks

### PR DESCRIPTION
**Pull request by jojonium**
## New

  - Bare-bones option menu that can be easily expanded. It includes options for "mute music" and "mute all" for now.
  - New placeholder sounds for almost everything that should have a sound effect. I know these are horrible but we can replace them with real ones eventually.

## Changed

  - Left and Right damage and size increases are now a function of the leftness/rightness of the shot, rather than being a flat cut-off.
  - The timer turns orange when 30 seconds are left and red when 10 seconds are left, plus it stops counting down after the hero dies.
  - Enemies with matryoshka > 1 drop low-level power ups when they die, as a reward for defeating a tough enemy

## Fixed

  - Fixed a bug that failed to recognize key presses when shift or caps lock was on (#76)

### CODE REVIEW

File: static/game/powerup.js [hunk 1]
```diff
@@ -4,15 +4,14 @@ import { Creature } from "./creature.js";
...
-
-// number of points a powerup is worth per magnitude
-const POINTS_FACTOR = 50;
 
 /**
  * @abstract
  */
 export class PowerUp extends Entity {
+  // number of points a powerup is worth per magnitude
+  static POINTS_FACTOR = 50;
```
Comment (Matt-Puentes) [inline static/game/powerup.js]: the "Static" keyword crashes on firefox

File: static/game/hero.js [hunk 2]
```diff
@@ -41,21 +41,44 @@ export class Hero extends Creature {
...
       for (let i = 0; i < 30; i++) {
-        // TODO move this to the destroy of powerup
         let randColor =
           "hsl(" + Math.floor(Math.random() * 360) + ", 100%, 50%)";
         const spark = new Particle(this.pos, randColor, EffectEnum.spark);
         spark.lineWidth = 15;
         spark.multiplier = 8;
         addParticle(spark);
       }
+      this.addPoints(entity.magnitude * PowerUp.POINTS_FACTOR);
```
Comment (Matt-Puentes) [inline static/game/hero.js]: This results in NaN when the "Static" keyword is removed from POINTS_FACTOR in powerup.js

File: static/game/block.js [hunk 3]
```diff
@@ -37,6 +38,7 @@ export function destroyBlock(cellVec, grantPoints = true) {
     if (grantPoints) {
       const gemType = blockField[cellVec.x][cellVec.y].gemType;
       if (gemType !== undefined) {
+        playSound("gem");
```
Comment (Matt-Puentes) [inline static/game/block.js]: maybe different sound effects for different gems?

File: static/game/enemy.js [hunk 4]
```diff
@@ -137,6 +138,13 @@ export class Enemy extends Creature {
         addToWorld(childEnemy);
         randDir += (2 * Math.PI) / spawnNum;
       }
+      if (this.matryoshka > 1) {
```
Comment (bandaloo) [inline static/game/enemy.js]: maybe this should be a chance in the future

File: static/game/timedisplay.js [hunk 5]
```diff
@@ -21,18 +21,16 @@ export class TimeDisplay extends GuiElement {
   }
 
   action() {
-    if (!getPause()) {
+    // prevent the timer from ticking down while the player is dead
```
Comment (bandaloo) [inline static/game/timedisplay.js]: thanks

File: static/game/timedisplay.js [hunk 6]
```diff
@@ -49,14 +47,17 @@ export class TimeDisplay extends GuiElement {
   }
 
   draw() {
+    let color = "white";
+    if (this.time < 100 * 30) color = "orange";
+    if (this.time < 100 * 10) color = "red";
```
Comment (bandaloo) [inline static/game/timedisplay.js]: cool

File: static/modules/buttons.js [hunk 7]
```diff
@@ -263,8 +263,8 @@ export function controlKeyupListener(e) {
 
   // is it a normal button?
   for (const nb of buttons.getButtons()) {
-    if (key === nb.key) {
-      e.preventDefault();
+    if (key === nb.key.toLowerCase()) {
+      if (!e.ctrlKey && !e.altKey && !e.metaKey) e.preventDefault();
```
Comment (bandaloo) [inline static/modules/buttons.js]: i suppose this works but it's not very DRY

File: static/game/hero.js [hunk 8]
```diff
@@ -41,21 +41,44 @@ export class Hero extends Creature {
...
+          break;
+        case 5:
+          magSound = "five";
+          break;
+      }
+      const puClassSound = entity.powerUpClass.toLowerCase().replace(" ", "-");
+      playSound("power-up");
+      const duration = getSound(puClassSound).duration;
+      setTimeout(() => playSound(puClassSound), 300);
+      setTimeout(() => playSound(magSound), duration * 1000 + 250);
```
Comment (bandaloo) [inline static/game/hero.js]: these magic numbers are a little confusing

File: static/game/block.js (hunk 3 above)
Comment (bandaloo) [reply static/game/block.js]: i like that idea

**Pull request merged**